drivers/mm/mm_drv_intel_adsp_mtl_tlb.c

```diff
@@ -62,15 +62,26 @@ DEVICE_MMIO_TOPLEVEL_STATIC(tlb_regs, DT_DRV_INST(0));
 	((ROUND_DOWN((hpsram_ebb_quantity) + 31u, 32u) / 32u) - 1u)
 
 #define L2_SRAM_PAGES_NUM			(L2_SRAM_SIZE / CONFIG_MM_DRV_PAGE_SIZE)
-#define MAX_EBB_BANKS_IN_SEGMENT	32
-#define SRAM_BANK_SIZE				(128 * 1024)
+#define MAX_EBB_BANKS_IN_SEGMENT		32
+#define SRAM_BANK_SIZE				KB(128)
 #define L2_SRAM_BANK_NUM			(L2_SRAM_SIZE / SRAM_BANK_SIZE)
-#define IS_BIT_SET(value, idx)		((value) & (1 << (idx)))
+#define IS_BIT_SET(value, idx)			((value) & (1 << (idx)))
+#define PMC_COMMUNICATION			defined(CONFIG_SOC_SERIES_INTEL_ACE) && \
+						defined(CONFIG_MISC_INTEL_COMM_WIDGET)
 
 static struct k_spinlock tlb_lock;
 extern struct k_spinlock sys_mm_drv_common_lock;
 
 static int hpsram_ref[L2_SRAM_BANK_NUM];
+#if PMC_COMMUNICATION
+#include <zephyr/drivers/misc/intel/comm_widget.h>
+
+#define PMC_COMMUNICATION_DEV		DT_PHANDLE_BY_IDX(DT_NODELABEL(tlb), pmc_communication, 0)
+
+static uint32_t used_pages;
+static uint32_t used_pages_reported;
```

Comment:
Terminology nitpick: these are counts of 64k "banks", right?  Not "pages"?  Xtensa MMU support is going to show up eventually and it's going to be really confusing if we have two different kinds of pages.